ledger-api/grpc-definitions/com/daml/ledger/api/v1/events_service.proto

```diff
@@ -0,0 +1,364 @@
+message EventProjection {
+  TransactionCommittedEventProjection transaction_committed_event_projection = 1;
+  ContractCreatedEventProjection contract_created_event_projection = 2;
+  ChoiceExercisedEventProjection exercised_event_projection = 4;
+}
+
+// No fine-grained projection available for now.
+message TransactionCommittedEventProjection {}
+
+// No fine-grained projection available for now. 
+message ContractCreatedEventProjection {
+  // TODO: think about the difference of filtering on stakeholders for "flat" Transactions
+  //       vs filtering on informees for TransactionTrees
+}
+
+// TODO: think about what kind of projections are required.
+message ChoiceExercisedEventProjection {
+  enum Kind { 
+    KIND_FULL = 0 ;          // corresponds to serving TransactionTrees
+    KIND_ARCHIVALS_ONLY = 1; // corresponds to serving "flat" Transactions
+  }
+
+  Kind kind = 1;
+}
+
+
+// Events: copied from v1.events.proto with light adjustments.
+//////////////////////////////////////////////////////////////
+
+message Event {
+    oneof kind {
+        TransactionCommittedEvent transaction_committed_event = 1;
+        ContractCreatedEvent contract_created_event = 2;
+        ChoiceExercisedEvent exercised_event = 4;
+    }
+}
+
+// Served as the first event of a transaction.
+// TODO:
+//  - filtering conditions
+//    - option 1: Visible to the submitters and informees of the transaction.
+//    - option 2: prefixed provided the stream contains an unfiltered event referring to this transaction
+//    - other options?
+//
+// Changes relative to v1.Transaction and v1.TransactionTree:
+//    - no need for `event_by_id` and `events`, as transaction boundary is communicated by TransactionOffset
+//    - no need for `root_event_ids`, as nesting structure is specified using `descendant_event_offset_upper_bound`
+//    - no need for `offset` as that is part of the `StreamEventsResponse`
+message TransactionCommittedEvent {
+
+  // A globally unique identifier useful to correlate transactions across different participant nodes.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string transaction_id = 1;
+
+  // The ID of the command which resulted in this transaction. Missing for everyone except the submitting party.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string command_id = 2;
+
+  // The workflow ID used in command submission. Only set if the ``workflow_id`` for the command was set.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string workflow_id = 3;
+
+  // Ledger effective time.
+  // Required
+  google.protobuf.Timestamp effective_at = 4;
+
+  // TODO: potentially missing
+  // string application_id = 6;
+  // repeated string submitters = 7; 
+}
+
+
+// Records that a contract has been created, and choices may now be exercised on it.
```

Comment:
If we want to output events in causal order, we should think about how to separate the announcement of "contract X is now active and can be exercised" from "this is the transaction that created this contract".